extensions/guacamole-auth-restrict/src/main/java/org/apache/guacamole/auth/restrict/RestrictionVerificationService.java

```diff
@@ -0,0 +1,337 @@
+        List<HostName> allowedHosts = HostRestrictionParser.parseHostList(allowedHostsString);
+        for (HostName hostName : allowedHosts) {
+            try {
+                // If the entry is an IP or Subnet, check the remote address against it directly
+                if (hostName.isAddress() && hostName.toAddress().contains(remoteHostName.asAddress()))
+                    return true;
+                
+                // Entry is a hostname, so resolve to IPs and check each one
+                for (IPAddress currAddr : hostName.toAllAddresses())
+                    if (currAddr.matches(remoteHostName.asAddressString()))
+                        return true;
+                
+            }
+            // If an entry cannot be resolved we will log a warning.
+            catch (UnknownHostException | HostNameException e) {
+                LOGGER.warn("Unknown host encountered in allowed host string: {}", hostName);
+                LOGGER.debug("Exception received trying to resolve host: {}", hostName, e);
+            }
+        }
+        
+        // If we've made it here, the allowed hosts do not contain the remote
+        // address, and the action should not be allowed;
+        return false;
+        
+    }
+    
+    /**
+     * Verifies the login restrictions supported by this extension for the user
+     * who is attempting to log in, throwing an exception if any of the
+     * restrictions result in the user not being allowed to log in.
+     * 
+     * @param context
+     *     The context of the user who is attempting to log in.
+     * 
+     * @param authenticatedUser
+     *     The AuthenticatedUser object associated with the user who is
+     *     attempting to log in.
+     * 
+     * @throws GuacamoleException 
+     *     If any of the restrictions should prevent the user from logging in.
+     */
+    public static void verifyLoginRestrictions(UserContext context,
+            AuthenticatedUser authenticatedUser) throws GuacamoleException {
+        
+        // Get user's attributes
+        Map<String, String> userAttributes = context.self().getAttributes();
+        String remoteAddress = authenticatedUser.getCredentials().getRemoteAddress();
+        
+        if (context.self().getEffectivePermissions().getSystemPermissions().hasPermission(SystemPermission.Type.ADMINISTER)) {
+            LOGGER.warn("User \"{}\" has System Administration permissions; additional restrictions will be bypassed.", authenticatedUser.getIdentifier());
+            return;
+        }
+            
+        // Verify time-based restrictions specific to the user
+        String allowedTimeString = userAttributes.get(RestrictUser.RESTRICT_TIME_ALLOWED_ATTRIBUTE_NAME);
+        String deniedTimeString = userAttributes.get(RestrictUser.RESTRICT_TIME_DENIED_ATTRIBUTE_NAME);
+        if ((allowedTimeString != null && !allowedTimeString.isEmpty() || (deniedTimeString != null && !deniedTimeString.isEmpty())))
```

Comment:
It doesn't look like all these null and empty checks are needed here - `allowedByTimeRestrictions()` does all the same checks. The same goes for the host checks (also in `verifyConnectionRestrictions()`).

Reply:
Yeah, I guess I should probably pick one place or the other to do the checks. I was trying to avoid calling the `verify` methods at all if they weren't present; however, I need to do the checks within those methods due to the logic of either rejecting or accepting the connection, so I should probably just skip the checks, here.

Reply:
Removed extra checks.